tests/pipeline/pipeline.cpp

```diff
@@ -0,0 +1,356 @@
+
+    const uint32_t inputValue = 90;
+    pipeInput.getPromise( "TestInputData" ).set( InputData( inputValue ));
+    pipeInput.execute();
+    convertPipeFilter.execute();
+    pipeOutput.execute();
+
+    const livre::UniqueFutureMap portFutures( pipeOutput.getPostconditions( ));
+    const OutputData& outputData =
+            portFutures.get< OutputData >( "TestOutputData" );
+    BOOST_CHECK_EQUAL( outputData.thanksForAllTheFish, 222 );
+}
+
+livre::Pipeline createPipeline( const uint32_t inputValue,
+                                size_t nConvertFilter = 1 )
+{
+    livre::Pipeline pipeline;
+    livre::PipeFilter pipeInput = pipeline.add< TestFilter >( "Producer" );
+    livre::PipeFilter pipeOutput = pipeline.add< TestFilter >( "Consumer" );
+
+    for( size_t i = 0; i < nConvertFilter; ++i )
+    {
+        std::stringstream name;
+        name << "Converter" << i;
+        livre::PipeFilter convertPipeFilter =
+                pipeline.add< ConvertFilter >( name.str( ));
+        pipeInput.connect( "TestOutputData", convertPipeFilter, "ConvertInputData" );
+        convertPipeFilter.connect( "ConvertOutputData", pipeOutput, "TestInputData" );
+    }
+
+    pipeInput.getPromise( "TestInputData" ).set( InputData( inputValue ));
+    return pipeline;
+}
+
+BOOST_AUTO_TEST_CASE( testSynchronousPipeline )
+{
+    const uint32_t inputValue = 90;
+
+    livre::Pipeline pipeline = createPipeline( inputValue, 1 );
+    pipeline.execute();
+
+    const livre::Executable& pipeOutput = pipeline.getExecutable( "Consumer" );
+    const livre::UniqueFutureMap portFutures( pipeOutput.getPostconditions( ));
+    const OutputData& outputData = portFutures.get< OutputData >( "TestOutputData" );
+    BOOST_CHECK_EQUAL( outputData.thanksForAllTheFish, 222 );
+}
+
+BOOST_AUTO_TEST_CASE( testWaitPipeline )
+{
+    const uint32_t inputValue = 90;
+    livre::Pipeline pipeline = createPipeline( inputValue, 1 );
+
+    livre::SimpleExecutor executor( 2 );
+    const livre::FutureMap pipelineFutures( pipeline.schedule( executor ));
+    pipelineFutures.wait();
+
+    const livre::Executable& pipeOutput = pipeline.getExecutable( "Consumer" );
+    const livre::UniqueFutureMap portFutures( pipeOutput.getPostconditions( ));
+    const OutputData& outputData = portFutures.get< OutputData >( "TestOutputData" );
+    BOOST_CHECK_EQUAL( outputData.thanksForAllTheFish, 222 );
+}
+
+BOOST_AUTO_TEST_CASE( testAsynchronousPipeline )
+{
+    const uint32_t inputValue = 90;
+
+    livre::Pipeline pipeline = createPipeline( inputValue, 1 );
+    livre::SimpleExecutor executor( 2 );
+
+    const livre::FutureMap pipelineFutures( pipeline.schedule( executor ));
+    const livre::Executable& pipeOutput = pipeline.getExecutable( "Consumer" );
+    const livre::UniqueFutureMap portFutures( pipeOutput.getPostconditions( ));
+    const OutputData& outputData = portFutures.get< OutputData >( "TestOutputData" );
+    BOOST_CHECK_EQUAL( outputData.thanksForAllTheFish, 222 );
```

Comment:
This code appears repeated several times below, the only thing that changes is the numbers that you use. Make a function for it?